viewer/packages/sector-parser/src/GltfSectorParser.ts

```diff
@@ -0,0 +1,176 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+import * as THREE from 'three';
+import { MeshGPUInstancing, InstancedMesh } from '@gltf-transform/extensions';
+import { Node, NodeIO, Document, Primitive, Mesh } from '@gltf-transform/core';
+import {
+  setBoxGeometry,
+  setConeGeometry,
+  setNutGeometry,
+  setQuadGeometry,
+  setTorusGeometry,
+  setTrapeziumGeometry
+} from './primitiveGeometries';
+
+//TODO: Move this utility our of core
+import { assertNever } from '../../../core/src/utilities';
+
+export enum RevealGeometryCollectionType {
+  BoxCollection,
+  CircleCollection,
+  ConeCollection,
+  EccentricConeCollection,
+  EllipsoidSegmentCollection,
+  GeneralCylinderCollection,
+  GeneralRingCollection,
+  QuadCollection,
+  TorusSegmentCollection,
+  TrapeziumCollection,
+  NutCollection,
+  TriangleMesh,
+  InstanceMesh
+}
+
+export default class GltfSectorParser {
+  private readonly _gltfReader: NodeIO;
+  constructor() {
+    this._gltfReader = new NodeIO();
+    this._gltfReader.registerExtensions([MeshGPUInstancing]);
+  }
+
+  public parseSector(data: ArrayBuffer) {
+    const document: Document = this._gltfReader.readBinary(data);
+
+    const defaultScene = document.getRoot().getDefaultScene();
+    const buffs: [RevealGeometryCollectionType, THREE.BufferGeometry][] = [];
+    defaultScene!.traverse((node: Node) => {
+      const res = this.processNode(node);
+      if (res) buffs.push(res);
+    });
+
+    return buffs;
+  }
+
+  private processNode(node: Node): [RevealGeometryCollectionType, THREE.BufferGeometry] | null {
+    const instanced = node.getExtension<InstancedMesh>(MeshGPUInstancing.EXTENSION_NAME)!;
+    const mesh = node.getMesh()!;
+
+    if (this.isEmptyNode(instanced, mesh)) return null;
```

Comment:
Avoid the ! and use a plain if to verify that instanced and mesh is not undefined/null, e.g.
```suggestion
    const instanced = node.getExtension<InstancedMesh>(MeshGPUInstancing.EXTENSION_NAME);
    const mesh = node.getMesh()!;

    if (!instanced && !mesh) return null;
```